tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/OpenTelemetrySpan.java

```diff
@@ -89,6 +95,35 @@ public Scope activate() {
         return new OpenTelemetryScope(delegate.makeCurrent());
     }
 
+    @Override
+    public io.helidon.tracing.Span baggage(String key, String value) {
+        Objects.requireNonNull(key, "Baggage Key cannot be null");
+        Objects.requireNonNull(value, "Baggage Value cannot be null");
+
+        // Check if OTEL Context is already available in Global Helidon Context.
+        // If not – use Current context.
+        Context context = Contexts.globalContext().get(Context.class)
+                .map(Context.class::cast)
+                .orElseGet(Context::current);
+
+        Baggage.builder()
+                .put(key, value)
+                .build()
+                .storeInContext(context)
+                .makeCurrent();
+        return (io.helidon.tracing.Span) delegate;
+    }
+
+    @Override
+    public String baggage(String key) {
+        // Check if OTEL Context is already available in Global Helidon Context.
+        // If not – use Current context.
+        Context context = Contexts.globalContext().get(Context.class)
```

Comment:
You should not be using `globalContext`, please use `Contexts.context().flatMap(it -> it.get(Context.class))`...

Reply:
Fixed!

Reply:
No, this was not fixed. You just moved it to another method and the code is still the same.

Reply:
yes, by bad, fixed.